### PR TITLE
Fix the release process

### PR DESCRIPTION
**Pull request by nagyv**
## Description

Check the changes if we update the release process as described in https://www.terraform.io/registry/providers/publishing

Closes https://github.com/gitlabhq/terraform-provider-gitlab/issues/896

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

**Comment by nagyv** [comment]
@timofurrer @armsnyder As I went through the Terraform provider publishing docs and updated the files you can see here, I think the only major difference is that we don't have a `terraform-registry-manifest.json` file.

The current, recommended release workflow is basically the same that we have. The GoReleaser configuration file differs in references to `terraform-registry-manifest.json`. Could this be the issue with the release process?

See https://www.terraform.io/registry/providers/publishing#terraform-registry-manifest-file

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -15,39 +15,35 @@ on:
...
+        uses: actions/checkout@v2.4.0
+      -
+        name: Unshallow
         run: git fetch --prune --unshallow
-      - uses: paultyng/ghaction-import-gpg@v2.1.0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14
```
Comment (armsnyder) [inline .github/workflows/release.yml:31]: Could you revert this line and use the go-version from the go-version job, so we use the same go-version in the release as we do in the build/test?

File: .goreleaser.yml [hunk 2]
```diff
@@ -1,5 +1,14 @@
+# Visit https://goreleaser.com for documentation on how to customize this
+# behavior.
+before:
+  hooks:
+    # this is just an example and not a requirement for provider building/publishing
```
Comment (armsnyder) [inline .goreleaser.yml:5]: Let's remove this hook since it's just an example

File: .goreleaser.yml [hunk 3]
```diff
@@ -24,17 +33,28 @@ archives:
 - format: zip
   name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
 checksum:
+  extra_files:
+    - glob: 'terraform-registry-manifest.json'
```
Comment (armsnyder) [inline .goreleaser.yml:37]: We would need to add this file to the root of the repository.

https://www.terraform.io/registry/providers/publishing#terraform-registry-manifest-file

> This file is conventionally named terraform-registry-manifest.json at the root of the provider repository and will be renamed in the release assets.

**Review by armsnyder — CHANGES_REQUESTED** [review]

**Comment by timofurrer** [comment]
I'm very curious what has changed that we are unable to release ... does it have to do with GitHub settings we made regarding protection of EE ?

File: .github/workflows/release.yml (hunk 1 above)
Comment (nagyv) [reply .github/workflows/release.yml]: Done.

File: .goreleaser.yml (hunk 2 above)
Comment (nagyv) [reply .goreleaser.yml]: Done.

File: .goreleaser.yml (hunk 3 above)
Comment (nagyv) [reply .goreleaser.yml:33]: I added the file.

**Comment by nagyv** [comment]
@timofurrer I think we did not change the release specific jobs. My guess is that we did not have the `terraform-registry-manifest.json` until now.

**Comment by nagyv** [comment]
@armsnyder I follow-up on your requests. Please, check and approve.

**Comment by timofurrer** [comment]
Can we actually build a `-pre` release to test this?

**Review by armsnyder — DISMISSED** [review]
Certainly worth trying, although to be clear I don't understand what went wrong in the first place. Did you see [my comment](https://github.com/gitlabhq/terraform-provider-gitlab/issues/896#issuecomment-1050549569) in the issue?

**Comment by timofurrer** [comment]
@nagyv I can't spot the release 👀

**Comment by nagyv** [comment]
> @nagyv I can't spot the release 👀

Neither do I. This release did not succeed either.

**Comment by timofurrer** [comment]
Just noticed, too. However, I don't think that it's related ;)

**Comment by timofurrer** [comment]
Worked: https://registry.terraform.io/providers/gitlabhq/gitlab/3.11.0-pre

Issue most likely was the missing manifest file (not sure why that appeared now 🤔 ) 

Also, the terraform upstream docs had a bug which I fixed here and in the upstream docs itself:

* https://github.com/hashicorp/terraform-website/pull/2161

**Pull request merged**